### PR TITLE
E2E Tests: minor improvements

### PR DESCRIPTION
**Pull request by brbrr**
#### Changes proposed in this Pull Request:

* add missing await call to onFailure handler.
* Log apache logs from tests container.
* Update slack reporter retry policy to fail fast.
* retry wpcom login if any of the steps failed.

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

n/a

#### Testing instructions:

* Make sure tests are green
*

### CODE REVIEW

File: projects/plugins/jetpack/tests/e2e/lib/env/playwright-environment.js [hunk 1]
```diff
@@ -121,9 +122,13 @@ class PlaywrightCustomEnvironment extends PlaywrightEnvironment {
...
 			const path = require( 'path' );
 			const filePath = path.resolve( `output/screenshots/${ fileName }` );
-			this.global.page.screenshot( { path: filePath } );
-
-			logger.slack( { type: 'file', message: filePath } );
+			try {
+				await this.global.page.screenshot( { path: filePath } );
+				logger.slack( { type: 'file', message: filePath } );
+			} catch ( error ) {
+				logger.debug( 'Failed to take screenshot due to: ' );
```
Comment (adimoldovan) [inline projects/plugins/jetpack/tests/e2e/lib/env/playwright-environment.js:129]: Shouldn't we use `logger.error` for this?

File: projects/plugins/jetpack/tests/e2e/lib/pages/wpcom/login.js [hunk 2]
```diff
@@ -27,25 +27,25 @@ export default class LoginPage extends Page {
...
-		await page.waitForSelector( passwordSelector, { state: 'visible', timeout: 30 } );
-		// Even if we wait for the field to become visible Playwright might still type the password too fast
-		// and the first characters will miss the password field. A short wait fixes this
-		await page.waitForTimeout( 2000 );
-		await page.type( passwordSelector, password );
-		await page.click( submitButtonSelector );
-
-		await this.page.waitForNavigation( { waitUntil: 'domcontentloaded' } );
-
 		try {
```
Comment (adimoldovan) [inline projects/plugins/jetpack/tests/e2e/lib/pages/wpcom/login.js:30]: I'm not a fan of these retry mechanisms, but I know that sometimes there's not much else left to do. I'm curious in this case if we know why the login sometimes fails and how often does it happen. Maybe we can find something else. Is there maybe a network request that's failing?

File: projects/plugins/jetpack/tests/e2e/lib/pages/wp-admin/jetpack.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ export default class JetpackPage extends Page {
 
 	async connect() {
 		const connectButtonSelector = '.jp-connect-full__button-container .dops-button';
-		return await page.click( connectButtonSelector );
+		return await page.click( connectButtonSelector, { timeout: 60000 } );
```
Comment (adimoldovan) [inline projects/plugins/jetpack/tests/e2e/lib/pages/wp-admin/jetpack.js:15]: Not really sure, but I think Jest times out before this. I think Jest has a 30s timeout and we shouldn't have PW timeout bigger than that.

**Review by adimoldovan — APPROVED** [review]
It looks good to me in general. I only left some suggestions.

File: projects/plugins/jetpack/tests/e2e/lib/pages/wp-admin/jetpack.js (hunk 3 above)
Comment (brbrr) [reply projects/plugins/jetpack/tests/e2e/lib/pages/wp-admin/jetpack.js:15]: I'm not sure about it. we have few more calls like this and I think it works as expected

File: projects/plugins/jetpack/tests/e2e/lib/pages/wpcom/login.js (hunk 2 above)
Comment (brbrr) [reply projects/plugins/jetpack/tests/e2e/lib/pages/wpcom/login.js:30]: yeah, I generally agree with you. Although, I'd like to improve the test overall stability, and then we can dig into the actual problem here.

File: projects/plugins/jetpack/tests/e2e/lib/env/playwright-environment.js (hunk 1 above)
Comment (brbrr) [reply projects/plugins/jetpack/tests/e2e/lib/env/playwright-environment.js:129]: yeah, makes sense

**Review by adimoldovan — APPROVED** [review]

**Review by adimoldovan — APPROVED** [review]

**Pull request merged**